obspy/taup/tau.py

```diff
@@ -31,6 +94,177 @@ def __str__(self):
+            (at the default location.) If a str, specify the location of the
+            legend. If you are plotting a single phase, you may consider using
+            the ``label_arrivals`` argument.
+        :type legend: bool or str
+        :param label_arrivals: Label the arrivals with their respective phase
+            names. This setting is only useful if you are plotting a single
+            phase as otherwise the names be large and possibly overlap or clip.
+            Consider using the ``legend`` parameter instead if you are plotting
+            multiple phases.
+        :type label_arrivals: bool
+        :param ax: Axes to plot to. If not given, a new figure with an axes
+            will be created. Must be a polar axes for the spherical plot and
+            a regular one for the cartesian plot.
+        :type ax: :class:`matplotlib.axes.Axes`
+        :param show: Show the plot.
+        :type show: bool
+
+        :returns: The (possibly created) axes instance.
+        :rtype: :class:`matplotlib.axes.Axes`
+        """
+        arrivals = []
+        for _i in self:
+            if _i.path is None:
+                continue
+            dist = _i.purist_distance % 360.0
+            if abs(dist - self.distance) / dist > 1E-5:
+                if plot_all is False:
+                    continue
+                # Mirror on axis.
+                _i = copy.deepcopy(_i)
+                _i.path["dist"] *= -1.0
+            arrivals.append(_i)
+        if not arrivals:
+            raise ValueError("Can only plot arrivals with calculated ray "
+                             "paths.")
+        discons = self.model.sMod.vMod.getDisconDepths()
+        if plot_type == "spherical":
+            if not ax:
+                plt.figure(figsize=(10, 10))
+                ax = plt.subplot(111, polar=True)
+            ax.set_theta_zero_location('N')
+            ax.set_theta_direction(-1)
+            ax.set_xticks([])
+            ax.set_yticks([])
+            intp = matplotlib.cbook.simple_linear_interpolation
+            radius = self.model.radiusOfEarth
+            for _i, ray in enumerate(arrivals):
+                # Requires interpolation otherwise diffracted phases look
+                # funny.
+                ax.plot(intp(ray.path["dist"], 100),
+                        radius - intp(ray.path["depth"], 100),
+                        color=COLORS[_i % len(COLORS)], label=ray.name,
+                        lw=2.0)
+            ax.set_yticks(radius - discons)
+            ax.xaxis.set_major_formatter(plt.NullFormatter())
+            ax.yaxis.set_major_formatter(plt.NullFormatter())
+            # Pretty earthquake marker.
+            ax.plot([0], [radius - arrivals[0].source_depth],
+                    marker="*", color="#FEF215", markersize=20, zorder=10,
+                    markeredgewidth=1.5, markeredgecolor="0.3", clip_on=False)
+
+            # Pretty station marker.
```

Comment:
FYI, I only change to annotations on the spherical side. The Cartesian wasn't going to run into the same problem.